### PR TITLE
KAFKA-12770: introduce `checkstyleVersion` build option (for overriding CheckStyle project-defined dependency version)

### PR DESCRIPTION
**Pull request by dejan2609**
@ijuma please review this.

**Rationale**: Requested by CheckStyle team (@romani); they needs this in order to add Apache Kafka project into their regression suit here https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/projects-to-test-on.properties

**JIRA ticket**: https://issues.apache.org/jira/browse/KAFKA-12770

**Related PR/comment:** https://github.com/apache/kafka/pull/10656#issuecomment-835809154

@romani how to use this (hopefully this will be merged into trunk at some point):
- this command uses project-defined CheckStyle version: `./gradlew checkstyleMain checkstyleTest` 
- while this one overrides it: `./gradlew checkstyleMain checkstyleTest -PcheckstyleVersion=8.41.1`

### CODE REVIEW

**Comment by dejan2609** [comment]
Note: rebased onto trunk (in order to pick up Gradle 6 -->> 7 version upgrade).

**Comment by dejan2609** [comment]
One flaky test failed, so I will retest this: 

```
Build / JDK 11 and Scala 2.13 / testCreateClusterAndCreateListDeleteTopic() – kafka.server.RaftClusterTest19s
Error
java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.UnknownServerException: The server experienced an unexpected error when processing the request.
```

**Comment by dejan2609** [comment]
retest this please

**Comment by romani** [comment]
as soon it is merged, we can add kafka to regression suite to make sure we will not leak regression on kafka code any more.

**Comment by dejan2609** [comment]
Roger that @romani ! I hope that @ijuma will review this PR when he finds some time.

Note: previous round of tests returned one error (flaky test, it seems) so I tried to trigger a new series of tests executions via `retest this please` comment but nothing happened...

File: README.md
Comment (showuon) [inline README.md]: question: do you think we should add some note in the above `Checkstyle` section?

**Review by showuon — COMMENTED** [review]
@dejan2609 , thanks for the PR. Overall LGTM. Left a minor comment. Thank you.

File: README.md
Comment (dejan2609) [reply README.md]: Sure @showuon ! Let me provide some more explanation here.

Comment (dejan2609) [reply README.md]: Ok, I just added few more details and force-pushed. 

I am going to resolve this conversation, but feel free to open a new one @showuon in case you want to suggest anything else).

**Comment by dejan2609** [comment]
@showuon thanx for a review, feel free to leave a comment after I polished a little bit (or open another review, any critique is welcomed !).

@ijuma I know you have a lot on your plate, but please take a moment for this PR: it is a prerequisite for CheckStyle team to include Kafka into their regression suite.

**Comment by dejan2609** [comment]
@ijuma all tests are in green. Can this be reviewed (and hopefully merged) ? 
Sorry for pushing, but [CheckStyle ](https://github.com/checkstyle/checkstyle) team is waiting for this.

**Comment by showuon** [comment]
@dejan2609 , sorry, I didn't make it clear. What I meant is [here](https://github.com/apache/kafka/blob/trunk/README.md#checkstyle), there's a `checkstyle` section describing what `checkstyle` is and how to run it. So I think we can add a note there to mention the `checkstyleVersion` argument can be passed to override default checkstyle version to do some experiments and regression testing.  What do you think?

**Comment by dejan2609** [comment]
Makes sense @showuon ! I totally understand your point: writing concise comments is a vital thing. 

I made some changes according to your suggestion, feel free to check (your input here is highly appreciated).

**Review by showuon — APPROVED** [review]
LGTM! Thanks.

**Comment by dejan2609** [comment]
If I may ask you @showuon: do we need to ping someone else for another review (or for merge into trunk) ? 

_Edit (just to answer to my self): there are two types of approvals and hence this PR does need an additional review(s)... in any case: thanx @showuon !_

**Comment by dejan2609** [comment]
CheckStyle team (@romani) needs this in order to add Kafka project into their regression suit here: 
https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/projects-to-test-on.properties

@showuon was so kind to provide his assistance, but we now need approval by someone who has a write access (@mumrah, @guozhangwang, @hachikuji, @ijuma, @junrao, @cmccabe or someone else).

Note: changes are small and simple (and risk doesn't exist).

**Comment by dejan2609** [comment]
Just tagging @ijuma here again (to bring back this PR up to the surface).

**Comment by dongjinleekr** [comment]
Cool. Working Like a charm.

`./gradlew checkstyleMain checkstyleTest`:

`./gradlew checkstyleMain checkstyleTest -PcheckstyleVersion=8.41.1`:

**Review by dongjinleekr — APPROVED** [review]

**Comment by dejan2609** [comment]
@ijuma  update: I already gather two approvals (made by fellow-contributors) for this PR, but I still need some maintainer to have his say.

**Comment by dejan2609** [comment]
Closed in favor of #10967; git commit is identical, but it seems that this PR  (with so many comments) is just out of sight.

**Pull request closed without merging**
